src/VisualStudio/Core/Def/Implementation/ChangeSignature/AddParameterDialogViewModel.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Notification;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ChangeSignature
+{
+    internal class AddParameterDialogViewModel : AbstractNotifyPropertyChanged
+    {
+        private readonly INotificationService? _notificationService;
+
+        public readonly Document Document;
+        public readonly int InsertPosition;
+
+        private readonly SemanticModel _semanticModel;
+
+        public AddParameterDialogViewModel(Document document, int insertPosition)
+        {
+            _notificationService = document.Project.Solution.Workspace.Services.GetService<INotificationService>();
+            _semanticModel = document.GetRequiredSemanticModelAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None);
+
+            Document = document;
+            InsertPosition = insertPosition;
+            ParameterName = string.Empty;
+            CallSiteValue = string.Empty;
+        }
+
+        public string ParameterName { get; set; }
+
+        public string CallSiteValue { get; set; }
+
+        private string TypeNameWithoutErrorIndicator
+        {
+            get
+            {
+                return TypeSymbol!.ToDisplayString();
+            }
+        }
+
+        private SymbolDisplayFormat _symbolDisplayFormat = new SymbolDisplayFormat(
+            genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+            miscellaneousOptions: SymbolDisplayMiscellaneousOptions.UseSpecialTypes);
+
+        public ITypeSymbol? TypeSymbol { get; set; }
+
+        public string TypeName
+        {
+            get
+            {
+                return (TypeSymbol!.IsErrorType() ? "(x) " : "") + TypeSymbol!.ToDisplayString(_symbolDisplayFormat);
+            }
+        }
+
+        internal void UpdateTypeSymbol(string typeName)
```

Comment:
Where is this called from? I kinda lost track in the PR review, but I'm guessing it's called from somewhere else? I would have assumed it was in TrySubmit or something like that.

Reply:
`OK_Click` in the `AddParameterDialog.xaml.cs` as of now. Thinking about moving it or binding it better.